### ISSUE TITLE
Cell breakpoints aren't always highlighted 

### ISSUE THREAD

**Issue by leonelgalan** [nteract/hydrogen#1602]
**Description:**

Breakpoints are only highlighted when _Untitled_ file is created after trying to open a _ipynb_. Saving this file to a _py_ removes all breakpoints highlights. For troubleshooting I positioned the cursor between `mark` and `down` on the first cell and triggered **"Editor: Log Cursor Scope"**.

**Steps to Reproduce:**

1. Click on _ipynb_ file in Tree View.

![Screen Shot 2019-04-25 at 4 19 01 PM](https://user-images.githubusercontent.com/727774/56766004-6502e080-6776-11e9-9216-939e100592d3.png)

Notice how `%%` and `%% markdown` are highlighted. Also **Python** is the selected grammar.

2. Save file with `.py` extension: foo.py

![Screen Shot 2019-04-25 at 4 19 25 PM](https://user-images.githubusercontent.com/727774/56766158-caef6800-6776-11e9-981e-93b68c4adc3e.png)

**Python** is still the selected grammar, but the highlights between cells are lost.

3. On its own the **Hydrogen breakpoints** appears to work:

![Screen Shot 2019-04-25 at 4 19 44 PM](https://user-images.githubusercontent.com/727774/56766280-0f7b0380-6777-11e9-8b20-05f35e7a4aa1.png)

**Versions:**

Which OS and which version of Hydrogen and Atom are you running?

* macOS 10.13.5
* Hydrogen 2.9.0

```
Atom    : 1.36.1
Electron: 2.0.18
Chrome  : 61.0.3163.100
Node    : 8.9.3
```

**Plugins:**

Do you have any Hydrogen plugins installed and active?

No Plugins.

**Logs:**

No warnings found on Atom's Developer Tools.

**Comment by elpollodiablo84** [comment]
I think I've found the problem: lately Atom switched to [tree-sitter](https://github.blog/2018-10-31-atoms-new-parsing-system/), a new parser to manage syntax highlighting. It's enabled by default from version 1.32. Disabling tree-sitter in "Settings > Core" solves this issue.
Sadly, I don't think you can inject an "old style" grammar into a new one, so I don't think it will be easy to definitively fix this problem.

**Comment by BenRussert** [comment]
@leonelgalan thanks for the detailed issue! 

If you wanted to dig into this (@leonelgalan, @elpollodiablo84, or anyone else looking), I would start with [our breakpoints.cson file](https://github.com/nteract/hydrogen/commits/master/grammars/breakpoints.cson). See [atom grammar creation docs](https://flight-manual.atom.io/hacking-atom/sections/creating-a-grammar/ ) for help with how this should work.

I have not looked into this deeply, so that is just a guess to start. Please comment if these dont end up helpful for finding the issue.

--- 

> 3. On its own the Hydrogen breakpoints appears to work

Can you clarify what you mean by "On its own"?

**Comment by lgeiger** [comment]
Here are more informations on why this is currently broken: https://github.com/atom/atom/issues/18196

**Comment by kylebarron** [comment]
I think you've diagnosed the problem well. I added the main Jupyter notebook code unless it's changed in the last few months.

When you open a Jupyter notebook that has metadata declaring it to be a Python file, Hydrogen forces Python syntax highlighting, which normally wouldn't appear until you save the file. This lets you run code through Hydrogen before saving the file. Apparently the Python syntax highlighting that's applied here by Hydrogen is always the TextMate grammar, even when the tree sitter global setting is enabled. Then when you save the file, Atom reapplies syntax highlighting and chooses the tree sitter one. 

So for consistent highlighting, the import-notebook code needs to accurately choose the right syntax highlighting to apply. 

Otherwise, not seeing breakpoints in a tree-sitter grammar is another issue.

**Comment by leonelgalan** [comment]
@BenRussert, I meant when choosing "Hydrogen breakpoints" manually.

@kylebarron I agree that this are separate issues. For me, the biggest is making it work with the new tree-sitter grammars. For the other issue, a workaround is saving the file.

I'm over my head, but I went through the docs: https://flight-manual.atom.io/hacking-atom/sections/creating-a-grammar/ and created [leonelgalan/tree-sitter-hydrogen-breakpoints](https://github.com/leonelgalan/tree-sitter-hydrogen-breakpoints) and attempted to contribute to this package following the instructions in the [Contributing Guide](https://github.com/nteract/hydrogen/blob/master/CONTRIBUTING.md).

It's still not working, not sure where it's failing, but I'll keep trying to make it work. 

### _tree-sitter-hydrogen-breakpoints/grammar.js_

I though the grammar could be very simple: the sequence of '%% followed by anything except end of line. I found the simplest grammars I could find([Aerijo/tree-sitter-todo](https://github.com/Aerijo/tree-sitter-todo/) and [tree-sitter/tree-sitter-embedded-template](https://github.com/tree-sitter/tree-sitter-embedded-template)) and build my own based on what I found there.

```js
module.exports = grammar({
  name: 'hydrogen_breakpoints',

  rules: {
    breakpoint: $ => repeat(
      $.directive
    ),

    directive: $ => seq(
      '%%',
      $._name
    ),

    _name: $ => /[^$]*/,
  }
});
```

### _hydrogen/grammars/breakpoints.cson_

```cson
scopeName: 'hydrogen.breakpoints'
name: 'Hydrogen breakpoints'
type: 'tree-sitter'
parser: 'tree-sitter-hydrogen-breakpoints'
scopes:
  'breakpoint': 'entity.name.function'
```

### _hydrogen/lib/main.js_

```js
const Hydrogen = {
  // ...

  activate() {
   // ...

    atom.grammars.addInjectionPoint('source.python', {
      type: 'comment.line',
      language (commentLine) {
        return 'hydrogen.breakpoints'
      },
      content (commentLine) {
        return commentLine;
      }
    });

   // ...
```

**Comment by kylebarron** [comment]
@leonelgalan I think it's unfixable pending https://github.com/atom/atom/issues/18196

**Comment by wadethestealth** [comment]
@kylebarron Yes, however I wouldn't call this a bug. There is support for TextMate Grammars. There is just a lack of support for TreeSitter Grammars atm. The only way to add current support for TreeSitter Grammars is to create our own grammar for each language (not ideal or worthwhile).
